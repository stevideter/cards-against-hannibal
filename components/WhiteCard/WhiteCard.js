import styles from "./WhiteCard.module.css";

const WhiteCard = (props) => (
  <div className={styles.card} id={props.id}>
    <p>{props.text}</p>
  </div>
);

export default WhiteCard;
