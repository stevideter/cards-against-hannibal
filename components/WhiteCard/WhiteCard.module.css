.card {
  background: white;
  color: black;
  width: 150px;
  height: 300px;
  padding: 10px;
  border-radius: 25px;
  border: 2px solid black;
  position: relative;
  margin: 10px;
}

.card p {
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  font-size: large;
  font-weight: bolder;
}