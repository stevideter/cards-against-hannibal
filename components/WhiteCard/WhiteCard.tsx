import styled from 'styled-components';

const StyledWhiteCard = styled.div`
background: white;
color: black;
width: 150px;
height: 300px;
padding: 10px;
border-radius: 25px;
border: 2px solid black;
position: relative;
margin: 10px;
`
const StyledP = styled.p`
margin: 0;
position: absolute;
top: 50%;
left: 50%;
transform: translate(-50%, -50%);
text-align: center;
font-size: large;
font-weight: bolder;
`
interface WhiteCardProps {
  id: string;
  text: string;
}
const WhiteCard = (props: WhiteCardProps) => (
  <StyledWhiteCard id={props.id}>
    <StyledP>{props.text}</StyledP>
  </StyledWhiteCard>
);

export default WhiteCard;
