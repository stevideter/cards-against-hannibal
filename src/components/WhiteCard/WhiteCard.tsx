import styled from 'styled-components';
import { Dispatch } from 'react';
import { GameBoardAction } from '../GameBoard/GameBoard';

const StyledWhiteCard = styled.div`
    background: white;
    box-shadow: rgb(102, 102, 102) 0px 2px 2px 0px;
    box-sizing: border-box;
    color: black;
    width: 10em;
    height: 10em;
    padding: 10px;
    border-radius: 10px;
    border: 2px solid black;
    position: relative;
    margin: 10px;
`;
const StyledP = styled.p`
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    font-size: 1.3em;
    font-family: MuseoSans, Arial, Helvetica, sans-serif;
    text-size-adjust: 100%;
`;
interface WhiteCardProps {
    id: string;
    text: string;
    dispatch: Dispatch<GameBoardAction>;
}
const WhiteCard: React.FunctionComponent<WhiteCardProps> = (
    props: WhiteCardProps
) => {
    function onClick(): void {
        props.dispatch({ type: 'playCard', payload: props.id });
    }
    return (
        <StyledWhiteCard id={props.id} onClick={onClick}>
            <StyledP>{props.text}</StyledP>
        </StyledWhiteCard>
    );
};

export default WhiteCard;
