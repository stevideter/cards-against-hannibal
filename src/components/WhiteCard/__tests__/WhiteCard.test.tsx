import WhiteCard from '../WhiteCard';
import renderer from 'react-test-renderer';

describe('WhiteCard', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<WhiteCard id="id" text="text" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
