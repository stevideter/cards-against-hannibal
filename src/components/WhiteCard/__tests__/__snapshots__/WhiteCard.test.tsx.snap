// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhiteCard renders correctly 1`] = `
<div
  className="sc-AxjAm hwiaAn"
  id="id"
  onClick={[Function]}
>
  <p
    className="sc-AxirZ URWBl"
  >
    text
  </p>
</div>
`;
